src/zap/zapimage.cpp

```diff
@@ -2177,7 +2177,7 @@ ZapImage::CompileStatus ZapImage::TryCompileMethodWorker(CORINFO_METHOD_HANDLE h
             // We skip the compilation of such methods and we don't want to
             // issue a warning or error
             //
-            if ((hrException == E_NOTIMPL) || (hrException == IDS_CLASSLOAD_GENERAL))
+            if ((hrException == E_NOTIMPL) || ((unsigned int)hrException == IDS_CLASSLOAD_GENERAL))
```

Comment:
A nit - can you please cast the constant to HRESULT instead?

Reply:
done